components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -34,6 +34,7 @@ abstract class EngineSession(
         fun onProgress(progress: Int) = Unit
         fun onLoadingStateChange(loading: Boolean) = Unit
         fun onNavigationStateChange(canGoBack: Boolean? = null, canGoForward: Boolean? = null) = Unit
+        fun onHistoryStateChange(historyList: List<HistoryItem>) = Unit
```

Comment:
I would be interesting to prototype this API for `browser-engine-system` too:
* Fire TV wants to use it (See #1340)
* The API is a bit different and implementing both makes sure we do not build something that only works for GeckoView.